complex/ft_msg.c

```diff
@@ -290,6 +309,48 @@ int ft_recv_dgram(void)
 	return -FI_ETIMEDOUT;
 }
 
+int ft_recv_dgram_flood(void)
+{
+	struct timespec s, e;
+	int credits, ret;
+	int64_t poll_time = 0;
+	int expected = ft.xfer_iter;
+	int got	     = 0;
+	do {
+		if (ft_rx.credits > (ft_rx.max_credits >> 1)) {
+			ret = ft_post_recv_bufs();
+			if (ret)
+				return ret;
+		}
+
+		credits = ft_rx.credits;
+
+		ret = ft_comp_rx();
+		if (credits != ft_rx.credits) {
+			int num_creds = ft_rx.credits - credits;
+			poll_time=0;
+			got+=num_creds;
+		}
+		if (ret)
+			return ret;
+
+		if(got >= expected) {
+			return 0;
```

Comment:
I was just looking at how to deal with this:

> 1) send immediate data.

This seems like the best option -- if only it would work on iwarp.  :/

> 2) use multiple receive buffers

This looks like it requires a fair amount of rework.  :/

> 3) use another endpoint

The receiver would need to inspect which EP each message arrived on, or look at multiple CQs.